### DISCUSSION TITLE
RTSP network recovery

### DISCUSSION THREAD

**Discussion by CyberCyclone** [TheWidlarzGroup/react-native-video#3851] [Q&A]
### What are you trying to do?

I'm trying to work out how to recover from a network outage. Currently what happens is that I have a RTSP stream that works fine, but if the stream drops out, it never reconnects and continues the stream. If I create the Video component before the stream is live, it won't start reading the stream once it doesn't come alive.

I couldn't find any props or methods that would help start / restart a stream.

Being a stream, I would expect that it would look at the URL and then start playing once the stream is live. I would also expect that if it drops out, it would auto-reconenct to the stream and start playing once it comes live.

Does anyone know if there is a way to achieve this?

**Comment by CyberCyclone** [comment]
It looks like this might be held back by this: https://github.com/google/ExoPlayer/issues/9689

**Comment by TWG-AI** [comment]
RTSP automatic reconnect after stream drop isn't built into react-native-video currently (the underlying ExoPlayer limitation is linked in the comments above). As a workaround, listen for the `onError` callback and remount the `<Video />` component by changing its `key` prop — this triggers a fresh connection attempt. You may also want a retry backoff loop so you don't hammer a stream that's legitimately down:

```js
const [retryKey, setRetryKey] = useState(0);

<Video
  key={retryKey}
  source={{ uri: rtspUrl }}
  onError={() => setTimeout(() => setRetryKey(k => k + 1), 3000)}
/>
```
